ReportSimulationOutput/measure.rb

```diff
@@ -2041,11 +2042,26 @@ def get_report_meter_data_timeseries(meter_names, unit_conv, unit_adder, timeser
     rows = timeseries_data['Rows']
     indexes = cols.each_index.select { |i| meter_names.include? cols[i]['Variable'] }
     vals = [0.0] * rows.size
+
+    # Calculate whether we need to shift values once up front
+    shift_values = {}
+    indexes.each_with_index do |_i, idx|
+      shift_values[idx] = false
+      if apply_ems_shift(timeseries_frequency)
+        if meter_names[idx].include? Constants::ObjectTypeWaterHeaterAdjustment
+          # Shift energy use adjustment to align with hot water energy use
+          shift_values[idx] = true
+        elsif meter_names[idx].include? Constants::ObjectTypePanHeater
+          # Shift energy use adjustment to align with HVAC operation and weather
+          shift_values[idx] = true
```

Comment:
I verified the timeseries output is correct; pan heater energy use only shows up when temperatures are below 32F.

I also refactored this code a bit because I realized we were previously calling `apply_ems_shift(timeseries_frequency)` inside two nested loops, which is very inefficient. So now we do everything up front.